src/simulate.jl

```diff
@@ -120,25 +120,31 @@ end
 Returns a simulation step function for the given `dt1` with stochastic elements, introducing possible events like vehicular needs and delayed returns.
 The function chooses between `simustep_alldrive`, `simustep_allplug`, or `simustep_event` probabilistically.
 """
-function get_simustep_stochastic(dt1::DateTime)
+function get_simustep_stochastic(dt1::DateTime, drive_starts_time, park_starts_time)
     date_part = Dates.Date(dt1)
-    dt_9am = DateTime(date_part, Dates.Time(9, 0, 0))
-    dt_5pm = DateTime(date_part, Dates.Time(17, 0, 0))
+    dt_9am = DateTime(date_part, drive_starts_time)
+    dt_5pm = DateTime(date_part, park_starts_time)
 
-    if dt_5pm - periodstep(1) ≤ dt1 < dt_5pm && rand() < prob_delayed_return
-        return simustep_base
+    rand_delayed_return = rand()
+    rand_event_return = rand()
+    rand_event = rand()
+
+    if dt_5pm - periodstep(1) ≤ dt1 < dt_5pm && rand_delayed_return < prob_delayed_return
+        delayed_return = 1 ## the event caused a later return than anticipated?
+        return simustep_base, (delayed_return, 0, 0)
     end
-    if rand() < prob_event_return
+    if rand_event_return < prob_event_return
+        event_return = 1 ## the event caused a return 
         if dt_9am - periodstep(1) ≤ dt1 < dt_5pm - periodstep(1)
-            return simustep_alldrive
+            return simustep_alldrive, (0, event_return, 0)
```

Comment:
Why did you add this extra info? It's potentially useful, but I don't see you using it anywhere, and it's going to mess up the other code.